### ISSUE TITLE
QuotaExceededError/missing pages in documentation

### ISSUE THREAD

**Issue by qoregexp** [dfahlander/Dexie.js#479]
I am trying to sort out possible issues with QuotaExceededError but none of the links here http://dexie.org/docs/DexieErrors/DexieError.html work (all get 404's).

I'm trying to cobble my application into working with iOS 9 without actually having access to an iOS 9 device.  My testers report that they can install small sets of data but not larger amounts, and I suspect that the quota is being exceeded but that this is not being reported.  Chrome gives the "Oh, snap" page.  I think Safari is not saying anything.  I thought that the browser was supposed to ask permission to extend the quota if necessary but this isn't happening.  

I also saw that indexedDB behaves differently regarding quotas depending on whether the store is to be persistent or temporary, but I don't see a way to specify this with Dexie.  Which is it actually using?

Thanks for any insight you can provide!

**Comment by nponiros** [comment]
You are right that some error pages are missing. There is an open issue for that.

Dexie uses the browser default when it comes to persistent or temporary storage. There is no way to change that currently. See also https://github.com/dfahlander/Dexie.js/issues/467 .

I'm not sure how well iOS9 actually works with indexedDB, I think it is pretty buggy. And chrome on iOS is in the end just safari but looks like chrome so it makes sense that it doesn't work. What kind of data are the users trying to add to indexedDB? From what I know iOS does not support blobs (also not iOS 10.2) which maybe is causing your issues. I also don't have an iOS9 device but if you provide a minimal example to reproduce the issue someone with the appropriate device can test it.

**Comment by qoregexp** [comment]
Thanks for your response.  The data is just numbers and strings, mainly.  I'm using localStorage for blobs.  But there is a lot of it -- it's basically large matrices and 20 MB for one matrix would not be unusual.

I'm not sure how buggy Safari indexedDB is in iOS 9 -- I thought it basically didn't work, but when I read up on the bugs and wrote tests to check for them, it all seemed to work for my testers.  They also report that they can install small matrics and the application works, so there may still be bugs, but they don't seem to affect my app.  

I'm concerned, though, that since Dexie uses the browser default, that behavior is going to be (more) unpredictable.  Maybe I just have to tell my users that even though it might seem like iOS 9 works, they're on their own with large datasets.

Here's a test case, but I don't know if the numbers are large enough to trigger anything.
[ixdbtest.zip](https://github.com/dfahlander/Dexie.js/files/819745/ixdbtest.zip)

**Comment by qoregexp** [comment]
Sorry, left out an include.
[ixdbtest.zip](https://github.com/dfahlander/Dexie.js/files/819757/ixdbtest.zip)

**Comment by kiranNegiloni** [comment]
@qoregexp Did you get a solution for Quota exceeded issue? Currently we are facing the Quota exceeded issue on Android 4.4.2(Hybrid application). We are trying to save around 20-25 MB data to indexDB.

Dexie version is 2.0.0-RC. I have noticed that whenever internal storage space is less than 800MB, users are facing this issue. Appreciate if anyone has the solution as users are facing issue in production environment.

**Comment by qoregexp** [comment]
No, I don't have a solution.  I have just been telling people that only very small data sets will work on iOS 9.  I would suspect similar issues with old versions of Android.  I still wish that Dexie had a way to specify permanent storage so that people would at least get a warning before their data was deleted (or I think that's the way it's supposed to work).  I had one customer whose data was gone every day when she started the app.  She had to do a backup when she finished, then the next day the data was gone and she would do a restore and continue working, do a backup, next data the data was gone again, etc.  Then one day this just stopped happening.  I don't remember her environment.  I suspect she was low on storage and perhaps deleted something else so that the low storage trigger stopped happening.

**Comment by bennyn** [comment]
**QuotaExceededError** is documented here https://dexie.org/docs/DexieErrors/Dexie.QuotaExceededError, so I guess the issue can be closed?
